### PR TITLE
Implement /dev/sbl_srv file

### PR DESCRIPTION
**Pull request by mikusp**
If there is a way to avoid making `dev` mod pub, let me know in suggestions.

### CODE REVIEW

**Comment by SuchAFuriousDeath** [comment]
We should probably initialize all the /dev devices somewhere inside the module.

File: src/kernel/src/fs/dev/sblsrv.rs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+/// An implementation of `/dev/sbl_srv`.
+#[derive(Debug)]
+pub struct SblService {}
+
+impl SblService {
+    pub const PATH: &'static VPath = vpath!("/dev/sbl_srv");
+
+    pub fn new() -> Arc<Self> {
+        let sbl_srv_cdevsw = Arc::new(CdevSw::new(
+            DriverFlags::from_bits_retain(0x80000000),
```
Comment (SuchAFuriousDeath) [inline src/kernel/src/fs/dev/sblsrv.rs:18]: This is DriverFlags::NEEDS_MINOR

**Review by SuchAFuriousDeath — CHANGES_REQUESTED** [review]

File: src/kernel/src/fs/dev/sblsrv.rs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+            Some(|_, _, _, _| Ok(())),
```
[the remark this answers is not in the document]
Comment (SuchAFuriousDeath) [reply src/kernel/src/fs/dev/sblsrv.rs:19]: By the way, we are pretty much categorically against stubbing like this. We always try to implement things as much as we can.

**Comment by ultimaweapon** [comment]
> We should probably initialize all the /dev devices somewhere inside the module.

Better to split each device into a separated module like `dev/dmem0` will be a responsible of `DmemManager`.

**Comment by ultimaweapon** [comment]
All of the devices in the `src/kernel/src/fs/dev` are legacy ones. It is a remnant of the old FS system. That means we are going to move all of them to the outside like `TtyManager`.

**Comment by SuchAFuriousDeath** [comment]
@ultimaweapon I think I already asked this somewhere, but you may have overlooked it. Aren't all the Cdevsws static in FreeBSD?

**Comment by ultimaweapon** [comment]
> @ultimaweapon I think I already asked this somewhere, but you may have overlooked it. Aren't all the Cdevsws static in FreeBSD? I don't really see a reason for Arc here

I can't remember why I was chosen for the current solution. But I already thought about static but decided to use Arc instead. Maybe we can try converted to a static version and see if we hit any problems.

File: src/kernel/src/fs/dev/sblsrv.rs (hunk 1 above)
Comment (mikusp) [reply src/kernel/src/fs/dev/sblsrv.rs]: NEEDS_MINOR is 0x00800000, according to FreeBSD that is D_INIT

**Comment by ultimaweapon** [comment]
@SuchAFuriousDeath can you check if your issues has been fixed?

**Comment by SuchAFuriousDeath** [comment]
Depends on whether you're fine with merging this
image

**Comment by ultimaweapon** [comment]
I think @mikusp already fixed that.

**Comment by SuchAFuriousDeath** [comment]
Oh :eyes:

File: src/kernel/src/fs/mod.rs [hunk 3]
```diff
@@ -25,7 +25,7 @@ pub use self::perm::*;
 pub use self::stat::*;
 pub use self::vnode::*;
 
-mod dev;
+pub mod dev;
```
Comment (ultimaweapon) [inline src/kernel/src/fs/mod.rs:28]: This should not need to be public anymore.

File: src/kernel/src/sbl/mod.rs [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+        let _ = make_dev(
+            &sbl_srv_cdevsw,
+            0,
+            "sbl_srv",
+            Uid::ROOT,
+            Gid::ROOT,
+            Mode::new(0o600).unwrap(),
+            None,
+            MakeDev::MAKEDEV_ETERNAL,
```
Comment (ultimaweapon) [inline src/kernel/src/sbl/mod.rs:31]: Is not this one is zero here?

**Review by ultimaweapon — CHANGES_REQUESTED** [review]

**Comment by SuchAFuriousDeath** [comment]
Superseded by #729

**Pull request closed without merging**